test/zc_weighted_vote_checkpoint.js

```diff
@@ -0,0 +1,409 @@
+    const tokenDetails = "This is equity type of issuance";
+    const decimals = 18;
+    
+    // Module key
+    const delegateManagerKey = 1;
+    const transferManagerKey = 2;
+    const stoKey = 3;
+    const checkpointKey = 4;
+    const burnKey = 5;
+
+    // Initial fee for ticker registry and security token registry
+    const initRegFee = new BN(web3.utils.toWei("1000"));
+
+    let currentTime;
+    const address_zero = "0x0000000000000000000000000000000000000000";
+    const one_address = "0x0000000000000000000000000000000000000001";
+
+    before(async () => {
+        currentTime = new BN(await latestTime());
+        account_polymath = accounts[0];
+        account_issuer = accounts[1];
+
+        token_owner = account_issuer;
+
+        account_investor1 = accounts[6];
+        account_investor2 = accounts[7];
+        account_investor3 = accounts[8];
+        account_investor4 = accounts[9];
+        account_temp = accounts[2];
+
+        // ----------- POLYMATH NETWORK Configuration ------------
+
+       // Step 1: Deploy the genral PM ecosystem
+       let instances = await setUpPolymathNetwork(account_polymath, token_owner);
+
+       [
+           I_PolymathRegistry,
+           I_PolyToken,
+           I_FeatureRegistry,
+           I_ModuleRegistry,
+           I_ModuleRegistryProxy,
+           I_MRProxied,
+           I_GeneralTransferManagerFactory,
+           I_STFactory,
+           I_SecurityTokenRegistry,
+           I_SecurityTokenRegistryProxy,
+           I_STRProxied,
+           I_STRGetter,
+           I_STGetter
+       ] = instances;
+
+
+        // STEP 4: Deploy the WeightedVoteCheckpoint
+        [I_WeightedVoteCheckpointFactory] = await deployWeightedVoteCheckpoint(account_polymath, I_MRProxied, 0);
+        [P_WeightedVoteCheckpointFactory] = await deployWeightedVoteCheckpoint(account_polymath, I_MRProxied, new BN(web3.utils.toWei("500")));
+
+        // Printing all the contract addresses
+        console.log(`
+        --------------------- Polymath Network Smart Contracts: ---------------------
+        PolymathRegistry:                  ${I_PolymathRegistry.address}
+        SecurityTokenRegistryProxy:        ${I_SecurityTokenRegistryProxy.address}
+        SecurityTokenRegistry:             ${I_SecurityTokenRegistry.address}
+        ModuleRegistry:                    ${I_ModuleRegistry.address}
+        ModuleRegistryProxy:               ${I_ModuleRegistryProxy.address}
+        FeatureRegistry:                   ${I_FeatureRegistry.address}
+
+        STFactory:                         ${I_STFactory.address}
+        GeneralTransferManagerFactory:     ${I_GeneralTransferManagerFactory.address}
+
+        WeightedVoteCheckpointFactory:     ${I_WeightedVoteCheckpointFactory.address}
+        -----------------------------------------------------------------------------
+        `);
+    });
+
+    describe("\t\t Test case for Weighted Vote Checkpoint module \n", async() => {
```

Comment:
Why are there new line and tab characters everywhere?

Reply:
It is only for better presentation